consensus.go

```diff
@@ -114,12 +114,19 @@ type Backend interface {
 	Hash(p []byte) []byte
 
 	// Init is used to signal the backend that a new round is going to start.
-	Init()
+	Init(*RoundInfo)
 
 	// IsStuck returns whether the pbft is stucked
 	IsStuck(num uint64) (uint64, bool)
 }
 
+// RoundInfo is the information about the round
+type RoundInfo struct {
+	IsProposer bool
+	Proposer   NodeID
```

Comment:
Do we need both `IsProposer` and `Proposer` fields? Looks like they are a bit redundant, although not sure about usage context, maybe it is easier to have both information in place...

Reply:
I think it does not hurt to have both, because the `Backend` implementation does not necessary need to have the local NodeID to check.

Reply:
Ok agreed!